### ISSUE TITLE
Transient failure loading index.html on staging

### ISSUE THREAD

**Issue by jonathancrabtree** [nih-cfde/dashboard#54]
Reported by @ACharbonneau at 1:52 PM EST 2/17/2021 on staging:

![portal-fail](https://user-images.githubusercontent.com/5394345/108368842-e982e680-71c8-11eb-8e9a-c65108362736.png)

Issue disappeared after refreshing a couple of times. May be due to transient HTTP error(s).

**Comment by jrchudy** [comment]
Looking at the browser console for this, I see this error reported:
> Access to XMLHttpRequest at 'https://app-dev.nih-cfde.org/dashboard-api/dcc_info' from origin 'https://app-staging.nih-cfde.org' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

@jonathancrabtree Is there a request hardcoded to interact with app-dev instead of relative to the server making the request?

**Comment by jonathancrabtree** [comment]
@jrchudy The UI expects the dashboard API URL and DERIVA/Chaise URL to be set correctly in config.js, and I don't think the auto-install procedure currently takes this into account. I'll modify it to do what you suggest, although I don't think this problem explains this particular issue, which definitely seemed to be transient in nature.

**Comment by jrchudy** [comment]
I think the above reported issue can still occur regardless of the issue I reported in a comment above. We should still move forward with implementing issue #55 to help make the request functionality more robust.

**Comment by mikedarcy** [comment]
I was able to capture this in the console when it happened. This is occurring because of an issue (in `webauthn`, I believe), which is causing HTTPD to core dump. When the HTTPD process serving this request dumps, the response back to the user is truncated. The ELB sees this and actually returns a 502 error. After the core dump the server actually recovers pretty quickly by spawning an new process. This error is _very_ intermittent, so it is not a major issue at the moment. We still need to figure it out though. The best remedy for this near term, until we can figure out the root cause, is to retry the XHR on a 502 a few times, maybe with a slight delay or backoff like we do in other places.

**Comment by mikedarcy** [comment]
Oh, I see you already proposed #55 - yeah that should deal with this issue nicely until we can get it squared away, and will also help for other situations if any of the peers in the ELB group go down or are hung up for whatever reason.

**Comment by ACharbonneau** [comment]
I’m getting intermittent submission failures with this message:

> This Flow has completed.
> Your CFDE submission (a7f6e5f9-67d8-42c2-809b-e46b24b8116a) failed to ingest into DERIVA with this error:
> Error ingesting to DERIVA: 502 Server Error: Bad Gateway for url: [https://app-staging.nih-cfde.org/ermrest/catalog/46/schema/CFDE/table/collection_assay_type/column/collection_id_namespace] Details: b’<html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n</body>\r\n</html>\r\n’

But can successfully submit the same catalog a few minutes later

**Comment by jonathancrabtree** [comment]
@ACharbonneau did you mean to file that last comment/bug report in cfde-deriva? It doesn't look like it has anything to do with dashboard, unless you were mentioning it to show that the 502 issue is widespread and persistent?

**Comment by jonathancrabtree** [comment]
#55 (auto-retry of failed AJAX requests with exponential backoff) is closed, so I don't think there's anything else to do here except see if it happens again (after that code is deployed)

**Comment by mikedarcy** [comment]
We've also enabled 502 retries in the deriva-py layer that the dashboard-api was using, so between that and #55 we should be covered while we try to find the root cause.

**Comment by ACharbonneau** [comment]
I think someone in the automation channel said the two were related so I stuck it here. I can make an issue in deriva as well if that's better

**Comment by jrchudy** [comment]
Reassigning to @ACharbonneau for Review/QA so this issue can be closed.

**Comment by ACharbonneau** [comment]
I haven't seen this happen in a while, so I think we're good. Will make a new issue if it comes back.
